src/eda.py

```diff
@@ -0,0 +1,208 @@
+        chosen_id_data = chosen_id_data.sort_values(by='Data', ascending=True)
+        
+        chosen_id_data['group'] = (~chosen_id_data['is_null']).cumsum().values
+
+        # Zgrupuj dane według 'group' i oblicz długość każdej sekwencji nulli w grupie
+        null_sequence_lengths = chosen_id_data.groupby('group').size() - 1
+
+        # Znajdź maksymalną długość ciągłej sekwencji nulli
+        nulls_in_row.loc[station]['max'] = null_sequence_lengths.max()
+        nulls_in_row.loc[station]['min'] = null_sequence_lengths.min()
+        nulls_in_row.loc[station]['mean'] = null_sequence_lengths.mean()
+        nulls_in_row.loc[station]['median'] = null_sequence_lengths.median()
+
+    chosen_stations = list(nulls_in_row[nulls_in_row['max'] <= threshold].index)
+    data_filtered = data[data.ID.isin(chosen_stations)]
+
+    return data_filtered
+
+def plot_nulls_percent(data, stations_start_end, region):
+    data['is_null'] = data['Suma dobowa opadów [mm]'].isnull()
+    for station in data['ID'].unique():
+        station_data = data[data['ID'] == station]
+        station_data = data[(pd.to_datetime(data['Data']) > pd.to_datetime(stations_start_end.loc[station]['Start_date'])) & (pd.to_datetime(data['Data']) < pd.to_datetime(stations_start_end.loc[station]['End_date']))]
+        stations_start_end.loc[station,'Null%'] = station_data['is_null'].sum()/len(station_data)
+
+    path = create_path_to_visualizations()
+    plt.figure(figsize=(10, 6))
+    sns.scatterplot(x='ID', y='Null%', data=stations_start_end, color='seagreen')
+    plt.xlabel('ID')
+    plt.ylabel('Null Percentage')
+    plt.title('Null Percentage for Each ID')
+    plt.xticks(rotation=45)  # Rotate x-axis labels for better readability
+    plt.grid(True)
+    plt.savefig(f"{path}/Null_percantage_station_{region}.png")
+
+def get_description(data):
+    print(data[["Suma dobowa opadów [mm]", "Status pomiaru SMDB", "Wysokość pokrywy śnieżnej [cm]", 
+    "Status pomiaru PKSN", "Wysokość świeżospadłego śniegu [cm]", "Status pomiaru HSS", "Status pomiaru GATS", "Status pomiaru RPSN"]].describe())
+
+def eda(region):
+    print("\nExploratory Data Analysis:\n")
+    data_path = create_path_to_data()
+    data_original = pd.read_csv(f'{data_path}/{region}_data.csv.gz', compression='gzip')
+
+    check_null_values(data_original)
+    
+    data = remove_duplicates(data_original)
+    data = change_types(data)
+
+    data = add_missing_rows(data)
+    data = fill_null_values(data)
+
+
+    stations_start_end = get_stations_start_end(data_original)
+    plot_nulls_percent(data, stations_start_end, region)
+
+    data_filtered = filter_by_number_of_nulls_in_row(data, stations_start_end, threshold=30)
+    data_filtered = interpolate_values(data_filtered)
+    data_filtered = correct_interpolate(data_filtered, stations_start_end)

```

Comment:
Can you add some descriptions to the functions? That would be great to see what was the problem and how interpolation is working in this case.

Reply:
Why does it need corrections? Please answer questions like that so it is easier to understand what you are doing.